oauthproxy.go

```diff
@@ -77,7 +77,7 @@ type OAuthProxy struct {
 	provider            providers.Provider
 	sessionStore        sessionsapi.SessionStore
 	ProxyPrefix         string
-	basicAuthValidator  basic.Validator
+	basicAuthValidator  func(string, string) bool
```

Comment:
I might work this out later but, why have we changed this to a function instead of the interface that was here before?

Reply:
Changed back to `basic.Validator`.